service/resource/vaultcrt/create_test.go

```diff
@@ -128,3 +129,50 @@ func Test_Resource_VaultCrt_newCreateChange(t *testing.T) {
 		}
 	}
 }
+
+func Test_Resource_VaultCrt_ensureVaultRole_Organizations_intact(t *testing.T) {
+	testCases := []struct {
+		CustomObject          certificatetpr.CustomObject
+		ExpectedOrganizations []string
+	}{
+		// Test 0 ensures a non-nil current state results in the create state to be
+		// empty.
+		{
+			CustomObject: certificatetpr.CustomObject{
+				Spec: certificatetpr.Spec{
+					ClusterID:        "foobar",
+					ClusterComponent: "api",
+					Organizations:    []string{"system:masters"},
+				},
+			},
+			ExpectedOrganizations: []string{"api", "system:masters"},
+		},
+	}
+
+	var err error
+	var newResource *Resource
+	{
+		resourceConfig := DefaultConfig()
+
+		resourceConfig.K8sClient = fake.NewSimpleClientset()
+		resourceConfig.Logger = microloggertest.New()
+		resourceConfig.VaultCrt = vaultcrttest.New()
+		resourceConfig.VaultRole = vaultroletest.New()
+
+		newResource, err = New(resourceConfig)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+	}
+
+	for i, tc := range testCases {
+		err := newResource.ensureVaultRole(tc.CustomObject)
+		if err != nil {
+			t.Fatal("case", i, "expected", nil, "got", err)
+		}
+		if !reflect.DeepEqual(tc.ExpectedOrganizations, key.Organizations(tc.CustomObject)) {
+			t.Fatalf("case %d expected %#v got %#v", i, tc.ExpectedOrganizations, key.Organizations(tc.CustomObject))
```

Comment:
This is the function that got the actual fix. I guess the test should be around this method. Also, when you want to test its behaviour because it was wrong before it should be called once and the result should be validated after using it. Calling it twice can still just produce the wrong behaviour without us noticing. So we should call it, set the result to a variable and use the variable for the inspection.

Reply:
I will try to make a test with the sort method
Will see this tomorrow